src/main/java/com/gotocompany/depot/http/response/HttpSinkResponse.java

```diff
@@ -10,30 +10,46 @@
 public class HttpSinkResponse {
 
     protected static final String SUCCESS_CODE_PATTERN = "^2.*";
-    private final HttpResponse response;
-
-    public HttpSinkResponse(HttpResponse response) {
-        this.response = response;
+    private boolean isFail;
+    private String responseCode;
+    private String responseBody;
+
+    public HttpSinkResponse(HttpResponse response) throws IOException {
+        setIsFail(response);
+        setResponseCode(response);
+        setResponseBody(response);
     }
 
-    public boolean isFailed() {
+    private void setIsFail(HttpResponse response) {
         if (response != null && response.getStatusLine() != null) {
-            return !Pattern.compile(SUCCESS_CODE_PATTERN).matcher(String.valueOf(response.getStatusLine().getStatusCode())).matches();
+            isFail = Pattern.compile(SUCCESS_CODE_PATTERN).matcher(String.valueOf(response.getStatusLine().getStatusCode())).matches();
         } else {
-            return true;
+            isFail = true;
         }
     }
 
-    public String getResponseCode() {
+    private void setResponseCode(HttpResponse response) {
         if (response != null && response.getStatusLine() != null) {
-            return Integer.toString(response.getStatusLine().getStatusCode());
+            responseCode = Integer.toString(response.getStatusLine().getStatusCode());
         } else {
-            return "null";
+            responseCode = "null";
         }
     }
 
-    public String getResponseBody() throws IOException {
+    private void setResponseBody(HttpResponse response) throws IOException {
         HttpEntity entity = response.getEntity();
-        return EntityUtils.toString(entity);
+        responseBody = EntityUtils.toString(entity);
```

Comment:
hmm. this might be more expensive that just consuming the entity. If we are not using the string, then it's better not to call toString.